experimenter/manifesttool/github_api.py

```diff
@@ -81,3 +84,47 @@ def _get_refs(repo: str, kind: str) -> list[Ref]:
         for page in paginated_api_request(f"repos/{repo}/{kind}")
         for ref in page
     ]
+
+
+@overload
+def fetch_file(repo: str, file_path: str, rev: str) -> str:
+    ...  # pragma: no cover
+
+
+@overload
+def fetch_file(repo: str, file_path: str, rev: str, download_path: Path) -> None:
+    ...  # pragma: no cover
```

Comment:
What's this for?

Reply:
Its just type hints because fetch_file has two modes: download to a path or return a string.

Reply:
But it can't infer that from the optional arg in the actual function implementation?  Like will type checking fail if you pull these out?

Reply:
The inferred type signature would otherwise be:

```
def fetch_file(
    repo: str,
    file_path: str,
    rev: str,
    download_path: Optional[Path] = None,
) -> Optional[str]
```

which does not capture the link between download_path and the return type.